Cython/Compiler/Nodes.py

```diff
@@ -5821,6 +5839,10 @@ def analyse_declarations(self, env):
             return
         else:
             self.lhs.analyse_target_declaration(env)
+            if (getattr(self.lhs, "entry", None) and
```

Comment:
What does it mean if we don't have an `entry` here? Will the attribute then be set later? Or does it somehow imply that we don't need it? Maybe we can make the node types explicit for which we need to do this?

Reply:
Usually just that the lhs variable isn't made purely from namenodes and attributes. Putting a breakpoint in there shows an example that comes from the utility code with:

```
AttributeNode(obj=IndexNode(...), attribute="name")
```

It just isn't useful to Cython as a recognised path to a standard library module so it implies that we don't need to try to identify it.

I'll take your suggestion of explicitly identifying the node types though (but it still needs the check for entry)